loader/get_environment.c

```diff
@@ -105,28 +106,39 @@ bool is_high_integrity() {
 }
 
 char *loader_getenv(const char *name, const struct loader_instance *inst) {
-    char *retVal;
-    DWORD valSize;
-
-    valSize = GetEnvironmentVariableA(name, NULL, 0);
+    int name_utf16_size = MultiByteToWideChar(CP_UTF8, 0, name, -1, NULL, 0);
+    if (name_utf16_size <= 0) {
+        return NULL;
+    }
+    wchar_t *name_utf16 = (wchar_t *)loader_stack_alloc(name_utf16_size * sizeof(wchar_t));
+    if (MultiByteToWideChar(CP_UTF8, 0, name, -1, name_utf16, name_utf16_size) != name_utf16_size) {
```

Comment:
I would like to take a bit of time to refactor the allocation code so that allocation calls here can be 1. Be able to use the allocation header and 2. Put in a single utility function.

Reply:
This code already uses allocation header. I think you mean `vk_loader_platform.h`. I moved `loader_stack_alloc` to `stack_allocation.h` in order to use it in `vk_loader_platform.h`. However, I do not know what I should put in a single utility function.